src/transformers/models/gptsan_japanese/modeling_gptsan_japanese.py

```diff
@@ -0,0 +1,1028 @@
+        layer_past: Optional[Tuple[torch.Tensor]] = None,
+        attention_mask: Optional[torch.FloatTensor] = None,
+        head_mask: Optional[torch.FloatTensor] = None,
+        use_cache: Optional[bool] = False,
+        output_attentions: Optional[bool] = False,
+    ) -> Tuple[Union[torch.Tensor, Tuple[torch.Tensor]], ...]:
+        r"""
+        Self-attention (if key_value_states is None) or attention over source sentence (provided by key_value_states).
+
+        Args:
+            hidden_states (`torch.Tensor`) :
+                [num_groups, tokens_per_group, hidden_dim] inputs to send to experts.
+            layer_past (`torch.Tensor`) :
+                past status from previous present output with using `use_cache`.
+            attention_mask (`torch.Tensor`) :
+                apply mask for attention.
+            head_mask (`torch.Tensor`) :
+                apply mask for heads.
+            use_cache (`bool`) :
+                output present key values.
+            output_attentions (`bool`) :
+                output attention probabirities.
+        Returns:
+            Tuple[torch.Tensor[num_groups, tokens_per_group, hidden_dim],...]
+        """
+        q, k, v = self._split2(hidden_states)  # [batch, sequence, 1, heads, kernel]
+
+        if use_cache is True:
+            present = (k, v)
+        # present shuld be ([batch, heads, sequence, hidden], [batch, heads, sequence, hidden])
+
+        if layer_past is not None:
+            pk, pv = torch.split(layer_past, [1, 1], dim=1)
+            pk = pk.squeeze(dim=1)
+            pv = pv.squeeze(dim=1)
+            if use_cache is True:
+                present = (pk, pv)
+            # pk, pv shuld be [batch, heads, sequence, hidden]
+            k = torch.cat([pk, k], dim=2)
+            v = torch.cat([pv, v], dim=2)
+
+        umask = attention_mask.unsqueeze(1)
+        scores = torch.einsum("bhsk,bhmk->bhsm", q, k)
+        scores *= self.d_kernel**-0.5
+        scores *= umask
+        scores -= (1 - umask) * 10000.0
+        probs = torch.exp(nn.functional.log_softmax(scores, dim=-1))  # same as mesh-tensorflow
+        if head_mask is not None:
+            probs = probs * head_mask
+        output = torch.einsum("bhsm,bhmk->bhsk", probs, v)  # [batch, heads, sequence, kernel]
+        output = output.transpose(1, 2)  # [batch, sequence, heads, kernel]
+        output = torch.einsum("bshk,hkc->bsc", output, self.o)  # [batch, sequence, hidden]
+
+        if use_cache is True:
+            outputs = (output, present)
+        else:
+            outputs = (output,)
+
+        if output_attentions:
+            outputs += (probs,)
+
+        return outputs
+
+
+class GPTSANJapaneseLayerSelfAttention(nn.Module):
+    """
+    Self Attention and Normalization Unit
+    """
+
+    def __init__(self, config, has_relative_attention_bias=False):
+        super().__init__()
+        self.SelfAttention = GPTSANJapaneseAttention(config)
```

Comment:
If you can avoid class names for variable, would be great
```suggestion
        self.self_attn = GPTSANJapaneseAttention(config)
```